hubploy/auth.py

```diff
@@ -182,24 +199,33 @@ def cluster_auth_gcloud(deployment, project, cluster, zone, service_key):
         'get-credentials', cluster
     ])
 
+    yield
+
 
 def cluster_auth_aws(deployment, project, cluster, zone, service_key):
     """
     Setup AWS authentication with service_key
 
     This changes *global machine state* on what current kubernetes cluster is!
     """
-    # move credentials to standard location
+
+    # Get credentials from standard location
     service_key_path = os.path.join(
         'deployments', deployment, 'secrets', service_key
     )
-    cred_dir = os.path.expanduser('~/.aws')
-    if not os.path.isdir(cred_dir):
-        os.mkdir(cred_dir)
-    shutil.copyfile(service_key_path, os.path.join(cred_dir, 'credentials'))
 
-    subprocess.check_call(['aws', 'eks', 'update-kubeconfig',
-                           '--name', cluster, '--region', zone])
+    # Set env variable for credential file location
+    os.environ["AWS_SHARED_CREDENTIALS_FILE"] = service_key_path
+
+    try:
+        subprocess.check_call(['aws', 'eks', 'update-kubeconfig',
```

Comment:
Actually, what we care about here is the correct path to the kubernetes config, without modifying the user's current kubernetes context. To that end, I propose the following:

1. Get a secure temporary file name from tempfile module
2. Set the `KUBECONFIG` environment variable to the temporary file name. 
3. Run the `update-kubeconfig` command. This will actually set up the file in the path `KUBECONFIG` points to.
4. All other commands (helm, etc) will use the path specified in `KUBECONFIG`
5. In our finally, we can unset `KUBECONFIG`, and delete the file.

This technique will actually work for google cloud (tested), and probably Azure too. 

What do you think of this?

Reply:
We should also save the current value of `KUBECONFIG` and restore it when done.

Reply:
Setting and unsetting the `KUBECONFIG` seems to be the same pattern we are using for the AWS credentials (which also avoids changing the user's settings, as you said), so I think that's a good idea. Keep everything working the same way for clarity (and stop messing with user files).

Seems like a straightforward set of changes, I'll work on it.

Reply:
Ah. The main problem I'm anticipating is that if I wanted to test and manage hubploy and the cluster locally, I'd need a manual `aws eks update-kubeconfig` call to do that, since the config location is reverted once hubploy is done working. That isn't a terribly difficult command to expect someone to run, though.

Reply:
@salvis2 yeah, you'd need to explicitly do that - this will leave your context unchanged. I think that's the 'right' behavior though.

Reply:
Yeah, I think that is the right thing to do - people will need to explicitly call that to change their context.